example/src/context/metamask.tsx

```diff
@@ -3,19 +3,22 @@ import React, {createContext, Dispatch, PropsWithChildren, Reducer, useReducer}
 
 export interface MetamaskState {
     isPolkadotSnapInstalled: boolean,
-    hasMetaMask: boolean
+    hasMetaMask: boolean,
+    hasPolkadotInstallFailed: boolean
```

Comment:
you don't need this(isPolkadotSnapInstalled will be false if it failed)
You need property that will hold installation error message